### ISSUE TITLE
require-selections does not check all possible types of union types

### ISSUE THREAD

**Issue by noahm** [graphql-hive/graphql-eslint#2963]
### Issue workflow progress

_Progress of the issue based on the
[Contributor Workflow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow)_

- [x] 1. The issue provides a reproduction available on GitHub, Stackblitz or CodeSandbox
- [x] 2. A failing test has been provided
- [x] 3. A local solution has been provided
- [x] 4. A pull request is pending review

---

**Describe the bug**

This feels related to #2318, only with inline fragments. On a union type where at least one selected type selects an available `id`, all other selected types are not all checked.

**To Reproduce** Steps to reproduce the behavior:
`npm run lint` in this [stackblitz](https://stackblitz.com/edit/github-whqdquez?file=operations%2Fquery.graphql&view=editor) summarized below:

```gql
type User {
  id: ID
  name: String
}

type Post {
  id: ID
  title: String
}

union UserOrPost = User | Post

type Query {
  user: User
  userOrPost: UserOrPost
}

query {
  userOrPost {
    ... on User { id }
    # no error reported, though inline fragment does not select `id` from Post
    ... on Post { title }
  }
}
```

**Expected behavior**

All available id fields are required to be selected from all types of a union that make one available

**Environment:**

- OS: Linux
- `@graphql-eslint/eslint-plugin`: 4.4.0
- Node.js: 22.22.3

**Additional context**

This is an edge case that we found at Twitch some years back and I added support for it to our custom gql validation tooling. I'd like to migrate off that custom stack to this eslint plugin, but this is a gap in coverage that we need to fix first. I'll be following this up with a PR shortly.
